lib/engine/game/g_18_oe/step/buy_sell_par_shares.rb

```diff
@@ -16,23 +26,101 @@ def actions(entity)
             actions << 'par' if can_ipo_any?(entity) || can_float_minor?(entity)
             actions << 'buy_company' if !purchasable_companies(entity).empty? || !buyable_bank_owned_companies(entity).empty?
             actions << 'sell_shares' if can_sell_any?(entity)
-
-            actions << 'pass' if !can_float_minor?(entity) && !bought? && !actions.empty?
+            actions << 'convert' if can_convert_any?
+            actions << 'pass' if !can_float_minor?(entity) && !actions.empty?
             actions
           end
 
+          def corporation_actions(entity)
+            return [] unless can_convert?(entity)
+
+            %w[convert pass]
+          end
+
           def can_buy_any_from_ipo?(entity)
             return unless @game.corporations.all?(&:ipoed)
 
             super
           end
 
+          def can_buy?(entity, bundle)
+            action = @round.current_actions.find { |a| a.is_a?(Engine::Action::Convert) }
+            return false if action && action&.entity != bundle.corporation
+
+            super
+          end
+
+          def can_sell?(entity, bundle)
+            action = @round.current_actions.find { |a| a.is_a?(Engine::Action::Convert) }
+            return unless bundle
+            return false if bundle.corporation.type == :regional
+            return false if action && action&.entity == bundle.corporation
+
+            super
+          end
+
           def can_float_minor?(entity)
             return unless entity.player?
 
             !bought? && entity.companies.any? { |company| @game.company_becomes_minor?(company) }
           end
 
+          def can_convert_any?
+            @game.corporations.select { |corp| can_convert?(corp) }.any?
```

Comment:
You're iterating through the array twice here.
```suggestion
            @game.corporations.any? { |corp| can_convert?(corp) }
```